tools/reactive-controllers/src/downstate.ts

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import type { LitElement, ReactiveController } from 'lit';
+
+export class DownState implements ReactiveController {
+    private host: LitElement;
+    private abortController: AbortController;
+
+    constructor(host: LitElement) {
+        this.host = host;
+        this.abortController = new AbortController();
+        this.host.addController(this);
+    }
+
+    public hostConnected(): void {
+        this.manage();
+    }
+
+    public hostDisconnected(): void {
+        this.removeEventListeners();
+        this.abortController.abort();
+    }
+
+    public manage(): void {
+        this.addEventListeners();
+    }
+
+    public unmanage(): void {
+        this.removeEventListeners();
+    }
+
+    addEventListeners(): void {
+        this.host?.addEventListener('pointerdown', this.handlePointerDown, {
+            signal: this.abortController.signal,
+        });
+        this.host?.addEventListener('pointerup', this.handlePointerUp, {
+            signal: this.abortController.signal,
+        });
+    }
+
+    removeEventListeners(): void {
+        this.host?.removeEventListener('pointerdown', this.handlePointerDown);
+        this.host?.removeEventListener('pointerup', this.handlePointerUp);
```

Comment:
We should remove the listeners with the abort signal.

Reply:
Already doing here
```js
 public hostDisconnected(): void {
        this.removeEventListeners();
        this.abortController.abort();
    }
```

Reply:
```js
removeEventListeners(): void {
        this.abortController.abort();
        this.host?.removeEventListener('pointerdown', this.handlePointerDown);
        this.host?.removeEventListener('pointerup', this.handlePointerUp);
    }
```

Reply:
The use of the abort controller here should prevent the need for `removeEventListener` code.